tyk-headless/templates/deployment-gw-repset.yaml

```diff
@@ -85,6 +85,8 @@ spec:
       - name: gateway-{{ .Chart.Name }}
         image: "{{ .Values.gateway.image.repository }}:{{ .Values.gateway.image.tag }}"
         imagePullPolicy: {{ .Values.gateway.image.pullPolicy }}
+        securityContext:
+            runAsNonRoot: true
```

Comment:
This only ensures that k8s won't allow the containers that are attempting to execute as UID 0. We also need to actually execute the container as a non-root user though.

E.g. with this added to the **Pod**-level `securityContext`:
```.yaml
runAsUser: 1000
fsGroup: 2000
```

This applies to all of the deployments in this PR.

For the gateway deployments, you'll also need to make sure that it tried to write the PID file and other files at an accessible location, e.g. like this in the tyk.conf:
```.json
"middleware_path": "/mnt/tyk-gateway/middleware",
"app_path": "/mnt/tyk-gateway/apps",
"pid_file_location": "/mnt/tyk-gateway/tyk.pid"
```

and a special ephemeral mount:
```.yaml
volumeMounts:
  ...
  - name: tyk-scratch
    mountPath: /mnt/tyk-gateway
...
volumes:
  ...
  - name: tyk-scratch
     emptyDir:
       medium: Memory
```

Now you can also add this to the container-level `securityContext`:
```.yaml
allowPrivilegeEscalation: false
privileged: false
readOnlyRootFilesystem: true
capabilities:
  drop:
    - all
```

There is a certain case where you'd want to allow root user execution (when running on a host network), but we don't support this case atm from what I see, so no need to worry about it yet.

Documentation for reference:
https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#podsecuritycontext-v1-core
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#securitycontext-v1-core

Reply:
Re `Memory` ephemeral medium: can be default on-disk, not necessarily in-memory, or even configurable through a value.

Reply:
@excieve Thanks for the review. Got the fixes in my last PR.